hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/JetSqlSerializerHook.java

```diff
@@ -350,12 +343,14 @@ public void afterFactoriesCreated(Map<Integer, DataSerializableFactory> factorie
         ArrayDataSerializableFactory mapDataFactory = (ArrayDataSerializableFactory) factories.get(SqlDataSerializerHook.F_ID);
 
         ConstructorFunction<Integer, IdentifiedDataSerializable>[] constructors =
-                new ConstructorFunction[SqlDataSerializerHook.TYPE_FIELD + 1];
+                new ConstructorFunction[SqlDataSerializerHook.LEN];
         constructors[SqlDataSerializerHook.MAPPING] = arg -> new Mapping();
         constructors[SqlDataSerializerHook.MAPPING_FIELD] = arg -> new MappingField();
         constructors[SqlDataSerializerHook.VIEW] = arg -> new View();
         constructors[SqlDataSerializerHook.TYPE] = arg -> new Type();
         constructors[SqlDataSerializerHook.TYPE_FIELD] = arg -> new Type.TypeField();
+        constructors[SqlDataSerializerHook.QUERY_DATA_TYPE] = arg -> new QueryDataType();
+        constructors[SqlDataSerializerHook.QUERY_DATA_TYPE_FIELD] = arg -> new QueryDataType.QueryDataTypeField();
```

Comment:
It might be less confusing this way:
```suggestion
        constructors[SqlDataSerializerHook.QUERY_DATA_TYPE] = arg -> new QueryDataType();
        // SqlDataSerializerHook.QUERY_ID
        constructors[SqlDataSerializerHook.MAPPING] = arg -> new Mapping();
        constructors[SqlDataSerializerHook.MAPPING_FIELD] = arg -> new MappingField();
        constructors[SqlDataSerializerHook.VIEW] = arg -> new View();
        constructors[SqlDataSerializerHook.TYPE] = arg -> new Type();
        constructors[SqlDataSerializerHook.TYPE_FIELD] = arg -> new Type.TypeField();
        // SqlDataSerializerHook.ROW_VALUE
        constructors[SqlDataSerializerHook.QUERY_DATA_TYPE_FIELD] = arg -> new QueryDataType.QueryDataTypeField();
```